src/maintenance/ClinicSteward.sol

```diff
@@ -172,6 +199,60 @@ contract ClinicSteward is IClinicSteward, RescuableBase, Multicall, AccessContro
     return IERC20(erc20Token).balanceOf(address(this));
   }
 
+  /* PRIVATE FUNCTIONS */
+
+  function _pullFunds(address asset, uint256 amount, bool useAToken) private {
+    _changePullLimit({asset: asset, amount: useAToken ? amount + 1 : amount, increasePullLimit: false});
+
+    if (useAToken) {
+      address aToken = POOL.getReserveAToken(asset);
+      // 1 wei surplus to account for rounding on multiple operations
+      ICollector(COLLECTOR).transfer(IERC20Col(aToken), address(this), amount + 1);
+      POOL.withdraw(asset, type(uint256).max, address(this));
+    } else {
+      ICollector(COLLECTOR).transfer(IERC20Col(asset), address(this), amount);
+    }
+  }
+
+  function _changePullLimit(address asset, uint256 amount, bool increasePullLimit) private {
+    uint256 assetPrice = IPriceOracleGetter(ORACLE).getAssetPrice(asset);
+
+    uint256 dollarAmount = (amount * assetPrice) / (10 ** IERC20Metadata(asset).decimals());
+
+    uint256 oldRestDollarPullLimit = restDollarPullLimit;
+
+    uint256 newRestDollarPullLimit;
+    if (increasePullLimit) {
+      newRestDollarPullLimit = oldRestDollarPullLimit + dollarAmount;
+    } else {
+      if (dollarAmount > oldRestDollarPullLimit) {
+        revert DollarPullLimitExceeded({
+          asset: asset,
+          assetAmount: amount,
+          dollarAmount: dollarAmount,
+          restDollarPullLimit: oldRestDollarPullLimit
+        });
+      }
+
+      newRestDollarPullLimit = oldRestDollarPullLimit - dollarAmount;
+    }
+
+    restDollarPullLimit = newRestDollarPullLimit;
+
+    emit DollarPullLimitChanged({oldValue: oldRestDollarPullLimit, newValue: newRestDollarPullLimit});
+  }
+
+  function _transferExcessToCollector(address asset, bool increasePullLimit) private {
```

Comment:
i think would remove the `increasePullLimit` functionality here.

It's inconsistent as if you liquidate `A` for `aA` the liquidated amount would become discounted.
Eventually, this does not need to be an exact science - the goal is that the guardian cannot overspend. If they can liquidate slightly less than the budget i think it's perfectly fine. Wdyt?